Support Databricks Runtime 11.3

GaryShen2008 opened the pull request:
Install ipykernel=6.12.1 when building for DB 11.3

GaryShen2008 committed:
Support Databricks Runtime 11.3

Signed-off-by: gashen <gashen@nvidia.com>

GaryShen2008 committed:
Install matplotlib for DB 11.3

Update build.sh to support DB 11.3
Signed-off-by: gashen <gashen@nvidia.com>

tgravescs commented on Databricks/Dockerfile:
```diff
@@ -59,6 +60,7 @@ RUN set -ex && \
         && /databricks/python3/bin/pip install \
             databricks-cli \
             ipython \
+        && if [ "$DATABRICKS_RUNTIME_VERSION" = "11.3" ]; then set -ex && /databricks/python3/bin/pip install ipykernel==6.12.1 matplotlib==3.4.3; fi \
```
so this requires us to deploy 2 different docker images right?  I'm wondering if we can do this dynamically in init script or something. We would need to look if we can get runtime fromthere

GaryShen2008 replied:
Yes, by this change, it'll ask us to deploy 2 different docker images.
Seems we only define one environment variable "DATABRICKS_RUNTIME_VERSION" to be 10.4 or 11.3.
I may verify if it can work by defining in the init script or the environment variables in advanced options.
If it can, we may be able to deploy only one. But the 10.4 image will install ipykernel with the new version and matplotlib, not sure if it'll be a problem on 10.4.

GaryShen2008 replied:
By removing the DATABRICKS_RUNTIME_VERSION from the Dockerfile, I verified the DB runtime version was actually decided by the version selected when creating the DB cluster.
So, no need to define this variable.
And it also makes we only deploy one image for both.

tgravescs replied:
you are just saying that ipykernel==6.12.1  works for both 10.4 and 11.3 then?  what version is 10.4 actually using?

GaryShen2008 replied:
There's no ipykernel installation in 10.4 from https://github.com/databricks/containers/blob/release-10.4-LTS/ubuntu/python/Dockerfile. It's added in 11.3-LTS.

GaryShen2008 committed:
Remove DATABRICKS_RUNTIME_VERSION

No need to define this variable
The version will follow the Databricks runtime version when creating a
DB cluster

Signed-off-by: gashen <gashen@nvidia.com>

GaryShen2008 commented:
@tgravescs can you review this PR? We have verified it works on both DB 10.4 and 11.3.

tgravescs commented on Databricks/Dockerfile:
```diff
@@ -59,6 +59,7 @@ RUN set -ex && \
         && /databricks/python3/bin/pip install \
             databricks-cli \
             ipython \
```
needs 2023 added to copyright

GaryShen2008 replied:
Updated.

tgravescs commented on Databricks/build.sh:
```diff
@@ -31,7 +31,6 @@ JAR_FILE=${JAR_FILE:-"rapids-4-spark_2.12-${JAR_VERSION}-cuda11.jar"}
 JAR_URL=${JAR_URL:-"${BASE_JAR_URL}/${VERSION}/${JAR_FILE}"}
```
needs copyright update

GaryShen2008 replied:
Updated.

tgravescs reviewed (commented):
looks fine other then copyrights

GaryShen2008 committed:
Update copyright

Signed-off-by: gashen <gashen@nvidia.com>

GaryShen2008 commented:
Updated copyright.
